packages/babel-parser/src/parser/error-message.js

```diff
@@ -129,18 +129,14 @@ export const ErrorMessages = Object.freeze({
   ParamDupe: "Argument name clash",
   PatternHasAccessor: "Object pattern can't contain getter or setter",
   PatternHasMethod: "Object pattern can't contain methods",
-  PipelineBodyNoArrow:
-    'Unexpected arrow "=>" after pipeline body; arrow function in pipeline body must be parenthesized',
-  PipelineBodySequenceExpression:
-    "Pipeline body may not be a comma-separated sequence expression",
-  PipelineHeadSequenceExpression:
-    "Pipeline head should not be a comma-separated sequence expression",
-  PipelineTopicUnused:
-    "Pipeline is in topic style but does not use topic reference",
-  PrimaryTopicNotAllowed:
-    "Topic reference was used in a lexical context without topic binding",
-  PrimaryTopicRequiresSmartPipeline:
-    "Primary Topic Reference found but pipelineOperator not passed 'smart' for 'proposal' option.",
+  PipeBodyCannotBeArrow:
+    'Unexpected arrow "=>" after pipeline body; arrow function acting as pipe body must be parenthesized due to operator precedence.',
+  PipeTopicRequiresHackPipes:
+    'Topic reference is used, but the pipelineOperator plugin was not passed a "proposal": "hack" or "smart" option.',
+  PipeTopicUnbound:
+    "Topic reference is unbound; it must be inside a Hack-style pipe body.",
+  PipeTopicUnused:
+    "Hack-style pipe body does not contain a topic reference; Hack-style pipes must use topic at least once.",
```

Comment:
Error codes are now part of the public API, so we should avoid changing them if possible.

Reply:
This has been addressed in 1c07c3ddc5f97734d02446be10816194dd72aa4c.